homeassistant/config.py

```diff
@@ -1015,21 +1029,114 @@ async def merge_packages_config(
     return config
 
 
+@callback
+def async_process_component_config_errors(
+    hass: HomeAssistant,
+    integration: Integration,
+    config_exception_info: list[ConfigExceptionInfo],
+    raise_on_failure: bool = False,
+) -> None:
+    """Process component configuration errors."""
+
+    p_ex: ConfigExceptionInfo
+    config_error_messages: list[tuple[str, ConfigExceptionInfo, str, str, str]] = []
+    general_error_messages: list[tuple[str, ConfigExceptionInfo]] = []
+    domain = integration.domain
+    for p_ex in config_exception_info:
+        link = p_ex.p_integration_link or integration.documentation
+        ex = p_ex.ex
+        p_name = p_ex.p_name
+        p_config = p_ex.p_config
+        if (log_message := p_ex.log_message) is None:
+            if TYPE_CHECKING:
+                assert p_name is not None and p_config is not None
```

Comment:
Why is this needed? If p_name and p_config can't be None, type annotations in `ConfigExceptionInfo` must be wrong?